phlib/mapimg.c

```diff
@@ -1720,8 +1720,10 @@ NTSTATUS PhGetMappedImageImportDll(
             while (TRUE)
             {
                 PhpMappedImageProbe(ImportDll->MappedImage, entry, sizeof(IMAGE_THUNK_DATA32));
+                IMAGE_THUNK_DATA32 entry_copy;
+                memcpy(&entry_copy, entry, sizeof(IMAGE_THUNK_DATA32));
```

Comment:
The code is fine and there's no point using a memcpy here with MSVC.

Reply:
MSVC does have an extension for unaligned pointers, but it's only if you use [__unaligned](https://learn.microsoft.com/en-us/cpp/cpp/unaligned?view=msvc-170). As far as standard C is concerned, this is currently UB.
I know on ARM (the non 64 bit) the original code would be terrible performance-wise as unaligned access is emulated by the kernel, but that architecture is fortunately dead and therefore won't ever be of any concern.
As with any UBs the real issue here is just that compilers are allowed to and will just do whatever. But granted, it does seem to function as intended on the current version of MSVC.

Reply:
(Just Curious) Because it's used only once, I doubt copying content is better than unaligned access. Marking as __unaligned is better than copying, isn't it?

Reply:
Does this work? @namazso 

```
typedef IMAGE_THUNK_DATA32 UNALIGNED* UNALIGNED_PIMAGE_THUNK_DATA32;
UNALIGNED_PIMAGE_THUNK_DATA32 entry;

entry = (UNALIGNED_PIMAGE_THUNK_DATA32)ImportDll->LookupTable;
```